Vizzuality/landgriffon
# (bugifix) GeoCoding: Add level param on getAdminRegion by name

### General description

This PR fixes a bug where we weren't filtering by AdminRegion level when it comes to getting a AdminRegion (and its GeoRegion) by a AdminRegion name.

I extended `getAdminRegionAndGeoRegionIdsByAdminRegionName` method's capabilities to optionally accept a options parameter which for now has a level property which, in case of be provided, will add a filtering by it

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

Create some AdminRegions with the same name and different level. Executing this method without providing a level end up with a random result. However, filtering it by level, you should get the matching one

We still have a uncovered use case where there are admin regions with SAME name AND level (will happen in real life)

- Apart from the added feature / bug fix, check overall performance, styling...

## Checklist before merging
